drivers/flash/flash_mcux_flexspi.c

```diff
@@ -0,0 +1,155 @@
+{
+	const struct flash_flexspi_config *config = dev->config;
+
+	FLEXSPI_UpdateLUT(config->base, index, cmd, count);
+
+	return 0;
+}
+
+int flash_flexspi_set_flash_config(const struct device *dev,
+		const flexspi_device_config_t *device_config,
+		flexspi_port_t port)
+{
+	const struct flash_flexspi_config *config = dev->config;
+	struct flash_flexspi_data *data = dev->data;
+
+	if (port >= kFLEXSPI_PortCount) {
+		LOG_ERR("Invalid port number");
+		return -EINVAL;
+	}
+
+	data->size[port] = device_config->flashSize * KB(1);
+
+	FLEXSPI_SetFlashConfig(config->base,
+			       (flexspi_device_config_t *) device_config,
+			       port);
+
+	return 0;
+}
+
+int flash_flexspi_reset(const struct device *dev)
+{
+	const struct flash_flexspi_config *config = dev->config;
+
+	FLEXSPI_SoftwareReset(config->base);
+
+	return 0;
+}
+
+int flash_flexspi_transfer(const struct device *dev,
+		flexspi_transfer_t *transfer)
+{
+	const struct flash_flexspi_config *config = dev->config;
+	status_t status = FLEXSPI_TransferBlocking(config->base, transfer);
+
+	if (status != kStatus_Success) {
+		LOG_ERR("Transfer error: %d", status);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+void *flash_flexspi_get_ahb_address(const struct device *dev,
+		flexspi_port_t port, off_t offset)
+{
+	const struct flash_flexspi_config *config = dev->config;
+	struct flash_flexspi_data *data = dev->data;
+	int i;
+
+	if (port >= kFLEXSPI_PortCount) {
+		LOG_ERR("Invalid port number");
+		return NULL;
+	}
+
+	for (i = 0; i < port; i++) {
+		offset += data->size[port];
+	}
+
+	return config->ahb_base + offset;
+}
+
+static int flash_flexspi_init(const struct device *dev)
+{
+	const struct flash_flexspi_config *config = dev->config;
+	flexspi_config_t flexspi_config;
+
+	if (config->xip) {
+		LOG_ERR("XIP not supported");
+		return -EINVAL;
+	}
+
+	FLEXSPI_GetDefaultConfig(&flexspi_config);
+
+	flexspi_config.ahbConfig.enableAHBBufferable = config->ahb_bufferable;
+	flexspi_config.ahbConfig.enableAHBCachable = config->ahb_cacheable;
+	flexspi_config.ahbConfig.enableAHBPrefetch = config->ahb_prefetch;
+	flexspi_config.ahbConfig.enableReadAddressOpt = config->ahb_read_addr_opt;
+	flexspi_config.enableCombination = config->combination_mode;
+	flexspi_config.rxSampleClock = config->rx_sample_clock;
+
+	FLEXSPI_Init(config->base, &flexspi_config);
+
+	return 0;
+}
+
+#define FLASH_FLEXSPI(n)						\
+	static const struct flash_flexspi_config			\
+		flash_flexspi_config_##n = {				\
+		.base = (FLEXSPI_Type *) DT_INST_REG_ADDR(n),		\
+		.ahb_base = (uint8_t *) DT_INST_PROP(n, ahb_base),	\
+		.ahb_bufferable = DT_INST_PROP(n, ahb_bufferable),	\
+		.ahb_cacheable = DT_INST_PROP(n, ahb_cacheable),	\
+		.ahb_prefetch = DT_INST_PROP(n, ahb_prefetch),		\
+		.ahb_read_addr_opt = DT_INST_PROP(n, ahb_read_addr_opt),\
+		.combination_mode = DT_INST_PROP(n, combination_mode),	\
+		.xip = DT_INST_PROP(n, xip),				\
+		.rx_sample_clock =					\
+			kFLEXSPI_ReadSampleClkLoopbackFromDqsPad,	\
```

Comment:
Option `kFLEXSPI_ReadSampleClkLoopbackFromDqsPad` require pin DQS, but this pin can be not available on some boards. So it is more flexible to use DTS for setup this option.

Reply:
I agree